pkg/core/consensus/round.go

```diff
@@ -334,11 +347,11 @@ func (c *Coordinator) CollectEvent(m bytes.Buffer) error {
 		lg.WithField("topic", topic).Debugln("storing future event")
 
 		// If it is a future agreement event, we store it on the
-		// corresponding round for step 1. This would mean that the
-		// queued agreement events for that round will be dispatched
-		// at the start of the first selection.
+		// `roundQueue`. This means that the event will be dispatched
+		// as soon as the Coordinator reaches the round in the event
+		// header.
 		if topic == topics.Agreement {
```

Comment:
I would be good to get rid of ad-hoc topic processing in the `Coordinator`, but I can live with this for now. Will create a separate issue